test/lint/lint-circular-dependencies.sh

```diff
@@ -26,6 +27,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "qt/transactiontablemodel -> qt/walletmodel -> qt/transactiontablemodel"
     "qt/walletmodel -> qt/walletmodeltransaction -> qt/walletmodel"
     "txmempool -> validation -> txmempool"
+    "util/validation -> validation -> util/validation"
```

Comment:
Is there a strong argument that justifies moving helper function `FormatStateMessage(const CValidationState &state)` to its own translation unit, whilst it depends on the translation unit it was moved from? 

I would prefer not to introduce new circular dependencies if they can be prevented.